tests/integration/hooks.go

```diff
@@ -42,6 +44,31 @@ var istioCrTearDown = func(ctx context.Context, sc *godog.Scenario, _ error) (co
 	return ctx, nil
 }
 
+var verifyIfControllerHasBeenRestarted = func(ctx context.Context, sc *godog.Scenario, _ error) (context.Context, error) {
+	c, err := testcontext.GetK8sClientFromContext(ctx)
+	if err != nil {
+		return ctx, err
+	}
+
+	podList := &v1c.PodList{}
+	err = c.List(ctx, podList, client.MatchingLabels{"app.kubernetes.io/component": "istio-operator.kyma-project.io"})
+	if err != nil {
+		return ctx, err
+	}
+	if len(podList.Items) < 1 {
+		return ctx, errors.New("Controller not found")
+	}
+
+	for _, cpod := range podList.Items {
+		if rc := cpod.Status.ContainerStatuses[0].RestartCount; rc > 0 {
```

Comment:
did you check if this is the correct counter, e.g. did you check if you change the spec and container is restarted this is the place it gets incremented?

Reply:
thats not good, what if container crashes, we need to be sure this is the right counter in case container crashes. Will try it out on my instance just to make sure.

Reply:
In addition to OOM it will check also for other potential issues that might happen to the controller and crash it like eg. runtime errors.

Reply:
this is the correct counter.